src/components/TableList/TableList.tsx

```diff
@@ -0,0 +1,103 @@
+import { useContext } from 'react';
+import { PersonComponent } from '../PersonComponent';
+import { Person } from '../../types/Person';
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from '../SearchLink';
+import { filterPeople } from '../../utils/filterPeople';
+import { sortPeople } from '../../utils/sortPeople';
+import { preparePeopleWithLinks } from '../../utils/preparePeopleWithLinks';
+import { TableContext } from '../../store/TableContextProvider';
+
+export const TABLE_FIELDS = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
+
+export const TableList = ({ people }: { people: Person[] }) => {
+  const { setIsEmptyMessage } = useContext(TableContext);
+  const [searchParams] = useSearchParams();
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+
+  let preparedPeople = preparePeopleWithLinks(people);
```

Comment:
Prepare people only once when you get them from server

Reply:
Here I am preparing the link itself for switching, and not directly the array with people